daft/file/video.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from daft.datatype import MediaType
+from daft.dependencies import av
+from daft.file import File
+from daft.file.typing import VideoMetadata
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+    import PIL
+
+    from daft.daft import PyFileReference
+    from daft.io import IOConfig
+
+
+class VideoFile(File):
+    """A video-specific file interface that provides video operations."""
+
+    @staticmethod
+    def _from_file_reference(reference: PyFileReference) -> VideoFile:
+        instance = VideoFile.__new__(VideoFile)
+        instance._inner = reference
+        return instance
+
+    def __init__(self, str_or_bytes: str | bytes, io_config: IOConfig | None = None) -> None:
+        if not av.module_available():
+            raise ImportError("The 'av' module is required to create video files.")
+        super().__init__(str_or_bytes, io_config, MediaType.video())
+
+    def __post_init__(self) -> None:
+        if not self.is_video():
+            raise ValueError(f"File {self} is not a video file")
+
+    def metadata(
+        self,
+        *,
+        probesize: str = "64k",
+        analyzeduration_us: int = 200_000,
```

Comment:
What kinds of values are acceptable for `probesize`? What does it do? Also, why not `probe_size`? Same questions for `analyzeduration_us`. I'm not sure a user would understand how to configure these options in a meaningful way without some additional documentation explaining what they do.

Reply:
This was per @everettVT's suggestion. I'm opting for removing them for now, and add additional metadata arguments at a later date.